reconcile/utils/ocm.py

```diff
@@ -28,6 +40,379 @@
+        SPEC_ATTR_PROVISION_SHARD_ID,
+        SPEC_ATTR_VERSION,
+        SPEC_ATTR_INITIAL_VERSION,
+    }
+
+    @staticmethod
+    def create_cluster(ocm: OCM, name: str, cluster: OCMSpec, dry_run: bool):
+        ocm_spec = OCMProductOsd._get_create_cluster_spec(name, cluster)
+        api = f"{CS_API_BASE}/v1/clusters"
+        params = {}
+        if dry_run:
+            params["dryRun"] = "true"
+
+        ocm._post(api, ocm_spec, params)
+
+    @staticmethod
+    def update_cluster(ocm: OCM, cluster_name: str, update_spec: Mapping[str, Any]):
+        ocm_spec = OCMProductOsd._get_update_cluster_spec(update_spec)
+        cluster_id = ocm.cluster_ids.get(cluster_name)
+        api = f"{CS_API_BASE}/v1/clusters/{cluster_id}"
+        params: dict[str, Any] = {}
+        ocm._patch(api, ocm_spec, params)
+
+    @staticmethod
+    def get_ocm_spec(
+        ocm: OCM, cluster: Mapping[str, Any], init_provision_shards: bool
+    ) -> OCMSpec:
+        if init_provision_shards:
+            provision_shard_id = ocm.get_provision_shard(cluster["id"])["id"]
+        else:
+            provision_shard_id = None
+
+        autoscale = cluster["nodes"].get("autoscale_compute")
+        autoscale_spec = (
+            OCMClusterAutoscale(
+                max_replicas=autoscale.get("max_replicas"),
+                min_replicas=autoscale.get("min_replicas"),
+            )
+            if autoscale
+            else None
+        )
+
+        spec = OSDClusterSpec(
+            product=cluster["product"]["id"],
+            id=cluster["id"],
+            external_id=cluster["external_id"],
+            provider=cluster["cloud_provider"]["id"],
+            region=cluster["region"]["id"],
+            channel=cluster["version"]["channel_group"],
+            version=cluster["openshift_version"],
+            multi_az=cluster["multi_az"],
+            instance_type=cluster["nodes"]["compute_machine_type"]["id"],
+            storage=cluster["storage_quota"]["value"] // BYTES_IN_GIGABYTE,
+            load_balancers=cluster["load_balancer_quota"],
+            private=cluster["api"]["listening"] == "internal",
+            disable_user_workload_monitoring=cluster[
+                "disable_user_workload_monitoring"
+            ],
+            provision_shard_id=provision_shard_id,
+            nodes=cluster["nodes"].get("compute"),
+            autoscale=autoscale_spec,
+        )
+
+        network = OCMClusterNetwork(
+            type=cluster["network"].get("type") or "OpenShiftSDN",
+            vpc=cluster["network"]["machine_cidr"],
+            service=cluster["network"]["service_cidr"],
+            pod=cluster["network"]["pod_cidr"],
+        )
+
+        ocm_spec = OCMSpec(
+            console_url=cluster["console"]["url"],
+            server_url=cluster["api"]["url"],
+            domain=cluster["dns"]["base_domain"],
+            spec=spec,
+            network=network,
+        )
+
+        return ocm_spec
+
+    @typing.no_type_check
```

Comment:
It would be helpful to have a comment for why this is set.

Reply:
I've changed the code a bit to enable type checking. In my opinion, this is a problem with how pydantic and mypy interact with Unions. In the end, I think it's better to have type checks even though we add the unnecessary isinstance check.